### ISSUE TITLE
Cannot Extract Nekopara Vol.2

### ISSUE THREAD

**Issue by Vacuumvoid** [xmoeproject/KrkrExtract#17]
Here is what I have done so far, please tell me if it's me doing something wrong or the program is misbehaving:
I have downloaded the 2 files: KrkrExtract.exe and KrkrExtract.dll 
I have placed these 2 files in the root directory of the Nekopara game.
https://i.gyazo.com/7dc434e1be679d3b43b459970aa5df6e.png

I then dragged nekopara_vol2.exe onto KrkrExtract.exe
The game then launches and no other windows pops up.
I am currently stuck on this stage and I could not extract any of the Nekopara files. Help is appreciated.

**Comment by 0xh3LL0** [comment]
after you download this repo, extract it somewhere else, since i afraid if you did it on program files directory you wont get the permission to write the data...
after you extract the repo, launch the krkrextract.exe and you'll found an empty window...
drag nekopara_vol2.exe on it and you'll have your nekorapa game launched, you can either minimize the game or close it...
leave behind it, should be a krkrextract window appear on your desktop, drop an xp3 file on it, and wait for the extraction...

good luck...

**Comment by xmoeproject** [comment]
I've tried the latest version, it works well.
Please check your anti-virus softwares, maybe they blocked KrkrExtract.
good luck

**Comment by feiyuhan** [comment]
I met the same error on extracting Nekopara Vol.2 .let me describe my situation. I drag the nekopara_vol2.exe to KrkrExtract.exe(I already put extract program in the root path。)the game and the extract program all launched correctly.But,when I click 选择(to choose xp3 file to extract)the extracting program dont have any responds and after seconds the program closed automatic﻿ly.

额，我用的时候，没有放在游戏根目录的时候会提醒放在根目录，然后放在根目录之后会出现，游戏程序直接拉到解压程序里面，然后游戏和解压程序是可以正常运行的，但是好像只要我点了解压程序里面的任何一个东西，像选择文件的按钮啥的，程序直接没响应了，过一会儿就自动关掉了。好像别人也是可以正常解压的，会是因为打了补丁的原因么？或者不是正版的原因（有可能会是这个原因吧？。），虽然我只是想拿这个游戏来尝试提取东西_(:з」∠)_

**Comment by feiyuhan** [comment]
I try it again,It works!  Previous operations is worry。when the extract program is launched correctly，and we should drag xp3 file to extract program and then it works.

好像是因为我之前点的选择是打包里面的选择，正常的情况只要正常启动之后，再把xp3的文件直接拖到解压的程序里面就会自动解压的。。忽然感觉是没有认真看文档的过？_(:з」∠)_

**Comment by kozalosev** [comment]
Hello! I've tried to extract files from the Steam version of Nekopara Vol. 1 and got the same problem: the game is running but nothing else appears.
After that fiasco, I repeated this operation on a non-Steam version and everything works fine!
So, maybe the extractor doesn't work for Steam versions?

_Many thanks for the program, by the way! It's a really cool thing!_

**Comment by marcussacana** [comment]
if your bug is just the window fully transparent...
I think you can try this:
- Open game proprieties in the steam menu
- Set Launch Options
- Write: -steam=false
- Launch the game
- Click at: Help
- Select: Display Advanced Settings
- Advanced Settings
- Uncheck the: Use Direct3D E-Mote Mode
- Restart the game

This can works if is just transparent window bug, and if you have perfomece problem try check the "Low Memory Perfomece Mode"

**Comment by kozalosev** [comment]
@marcussacana
Nope. It doesn't work. What's the difference between this issue and that one I referred to in the previous comment? You answered there yesterday that Steam version has the CEG Protection that's need to be cracked first. What's wrong?

**Comment by marcussacana** [comment]
i had problem with transparent window before, my some reason in the nekopara scripts have when the game start he check if the steam is enabled (paramter -steam=false force disable) and if is enable he hidden 2 options in the Advanced Menu...
The Transparent window maybe is a problem with render the game, disabling the Direct3D can help fix the transparent window bug.

The CEG Protection he is who force you login into steam before open a game...
He "comunicate" with the steam process checking if you lauch launch the game from the steam menu, if not he says to the steam start the game and close the process started by you...
With this when the KrKrExtract launch the game the CEG protection detect you launch the game out of the steam and says to the steam start the game, after this the process started by the KrKrExtract automatically close and you can't use the KrKrExtract...
Cracking the CEG you can run the game without he restart.

If your transparent window fix with the crack, is because a steam, literally... disable some resource of the game that needs to run in your machine... i found this code only in the nekopara.
